examples/server/server.cpp

```diff
@@ -529,8 +541,10 @@ struct llama_server_context
 
     std::vector<task_server> queue_tasks;
     std::vector<task_result> queue_results;
-    std::mutex mutex_tasks;
+    std::vector<task_multi>  queue_multitasks;
+    std::mutex mutex_tasks; // also guards id_gen
     std::mutex mutex_results;
+    std::mutex mutex_multitasks;
```

Comment:
I don't think we need a new mutex - try to reuse the existing `mutex_tasks`.
It should guard both `queue_tasks` and `queue_multitasks`

Reply:
Done, and retested.